Welcome modal on logged-out homepage

abenzer opened the pull request:
Shows a soft welcome modal to logged-out visitors on the homepage.

mozzius commented on src/components/WelcomeModal.tsx:
```diff
@@ -0,0 +1,312 @@
+  )
+}
+
+const styles = StyleSheet.create({
+  modalOverlay: {
+    ...a.fixed,
+    ...a.inset_0,
+    ...a.justify_center,
+    ...a.align_center,
+    zIndex: 9999,
+    backgroundColor: 'rgba(0,0,0,0.2)',
+    ...web({
+      backdropFilter: 'blur(15px)',
+    }),
+  },
```
rather than using `StyleSheet`, we usually just inline the styles like this:

```jsx
<View style={[a.fixed, a.inset_0, a.justify_center, a.align_center, a.z_50, {backgroundColor: 'rgba(0,0,0,0.2)'} />
```

mozzius commented on src/components/WelcomeModal.tsx:
```diff
@@ -0,0 +1,312 @@
+  const [exploreLinkHovered, setExploreLinkHovered] = React.useState(false)
+  const [signInLinkHovered, setSignInLinkHovered] = React.useState(false)
+
+  React.useEffect(() => {
+    Animated.timing(fadeAnim, {
+      toValue: 1,
+      duration: 100,
+      useNativeDriver: true,
+    }).start()
+  }, [fadeAnim])
+
+  const fadeOutAndClose = React.useCallback(
+    (callback?: () => void) => {
+      Animated.timing(fadeAnim, {
+        toValue: 0,
+        duration: 100,
+        useNativeDriver: true,
+      }).start(() => {
+        control.close()
+        if (callback) callback()
+      })
+    },
+    [fadeAnim, control],
+  )
```
We typically avoid React Native's `Animated` API - since this is web only, we can just use CSS

mozzius commented on src/components/hooks/useWelcomeModal.ts:
```diff
@@ -0,0 +1,41 @@
+    if (
+      isWeb &&
+      !hasSession &&
+      sessionStorage.getItem('welcomeModalShown') !== 'true' &&
```
decided to use the SessionStorage API so it doesn't immediately pop up again if you reload. However it will show again in a new tab

mozzius reviewed (approved):
Looks good! Looking forward to seeing how the experiment performs

Pushed up some tweaks:
- aligned code with our typical style (i.e. inlined styles)
- replaced JS animation with CSS one
- fix a11y (add focus guards/focus scope, add `aria-modal` etc)

abenzer commented:
@mozzius thanks for the tweaks! i made a few more:

1. fixed a mismatch in text color between the "Sign in" and "Explore the app" button
2. reduced the gap between the two buttons to match the Figma design
3. updated your logic of only showing the modal once per session → showing the modal until the user closes it
